easycv/apis/export.py

```diff
@@ -145,31 +153,127 @@ def _export_cls(model, cfg, filename):
 
 def _export_yolox(model, cfg, filename):
     """ export cls (cls & metric learning)model and preprocess config
-
     Args:
         model (nn.Module):  model to be exported
         cfg: Config object
         filename (str): filename to save exported models
     """
-    if hasattr(cfg, 'test_pipeline'):
-        # with last pipeline Collect
-        test_pipeline = cfg.test_pipeline
-        print(test_pipeline)
-    else:
-        print('test_pipeline not found, using default preprocessing!')
-        raise ValueError('export model config without test_pipeline')
 
-    config = dict(
-        model=cfg.model,
-        test_pipeline=test_pipeline,
-        CLASSES=cfg.CLASSES,
-    )
+    if hasattr(cfg, 'export') and (getattr(cfg.export, 'use_jit', False) or
+                                   getattr(cfg.export, 'export_blade', False)):
+        device = 'cuda' if torch.cuda.is_available() else 'cpu'
+
+        model = copy.deepcopy(model)
+        model.eval()
+        model.to(device)
+
+        end2end = cfg.export.get('end2end', False)
+        if LooseVersion(torch.__version__) < LooseVersion('1.7.0') and end2end:
+            raise ValueError('`end2end` only support torch1.7.0 and later!')
+
+        batch_size = cfg.export.get('batch_size', 1)
+        img_scale = cfg.get('img_scale', (640, 640))
+        assert (
+            len(img_scale) == 2
+        ), 'Export YoloX predictor config contains img_scale must be (int, int) tuple!'
+
+        input = 255 * torch.rand((batch_size, 3) + img_scale)
+
+        if hasattr(cfg, 'test_pipeline'):
+            # with last pipeline Collect
+            test_pipeline = cfg.test_pipeline
+            print(test_pipeline)
+        else:
+            print('test_pipeline not found, using default preprocessing!')
+            raise ValueError('export model config without test_pipeline')
+
+        target_size, keep_ratio, pad_val, mean, std, to_rgb = parse_pipleline(
```

Comment:
remove `parse_pipleline`, user's pipeline is varied, and this should be performed according to the user's behavior.